sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -18,18 +20,32 @@
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
-import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Objects;
 
 /**
  * This class is an abstraction over many different ways that binary data can be represented. The {@link BinaryData}
- * can be created from {@link InputStream}, {@link Flux} of {@link ByteBuffer}, {@link String}, {@link Object} and byte
- * array. The data is collected from provided sources and stored into a byte array.
+ * can be created from {@link InputStream}, {@link Flux} of {@link ByteBuffer}, {@link String}, {@link Object}, or byte
+ * array.
+ * <p><strong>Immutable data</strong></p>
+ * {@link BinaryData} is constructed by coping the given data. Once {@link BinaryData} is instantiated, it's data can
+ * not be changed. It provides various convenient APIs to get data out of {@link BinaryData}, they all starts with the
+ * <b>BinaryData.to...()</b>, For example {@link BinaryData#toBytes()}.
  * <p>
- * It also provides a way to serialize and deserialize an {@link Object} into {@link BinaryData} given an
- * {@link ObjectSerializer}. Code samples are explained below.
+ * It provides a way to serialize {@link Object} into {@link BinaryData} using
+ * {@link BinaryData#fromObject(Object, ObjectSerializer)} where you can provide your {@link ObjectSerializer}.
+ * <p>
+ * It provides a way to de-serialize {@link BinaryData} into specified {@link Object} using
+ * {@link BinaryData#toObject(Class, ObjectSerializer)} where you can provide object type and your
+ * {@link ObjectSerializer}.
+ * <p>
+ * It provides API to use default json serializer which is available in classpath. The serializer on classpath must
+ * implement {@link JsonSerializer} interface.
+ * <p>Read more about <a href="http://aka.ms/azsdk/java/wiki/serialization">serialization</a>.
```

Comment:
This seems to just be restating the API in the class, and is probably unnecessary. I would focus more on pointing users to the appropriate JavaDoc documentation, e.g. JsonSerializer, etc